### PR TITLE
feat(FEC-10835): expose share to plugin

### PR DESCRIPTION
**Pull request by Yuvalke**
### Description of the Changes

Issue: share configured ui from kaltura player.
Solution: handle this config as plugin config.

### CheckLists

- [ ] changes have been done against master branch, and PR does not conflict
- [ ] new unit / functional tests have been added (whenever applicable)
- [ ] test are passing in local environment
- [ ] Travis tests are passing (or test results are not worse than on master branch :))
- [ ] Docs have been updated

### CODE REVIEW

File: src/kaltura-player.js [hunk 1]
```diff
@@ -147,7 +147,6 @@ class KalturaPlayer extends FakeEventTarget {
           }
         )
         .then(mediaConfig => {
```
Comment (Yuvalke) [inline src/kaltura-player.js:149]: change to resolve without function

File: src/kaltura-player.js [hunk 2]
```diff
@@ -278,11 +277,6 @@ class KalturaPlayer extends FakeEventTarget {
       delete localPlayerConfig.sources;
     }
     this._localPlayer.configure(localPlayerConfig);
-    const uiConfig = config.ui;
-    if (uiConfig) {
-      this._configEvaluator.evaluateUIConfig(uiConfig, this.config);
-      this._uiWrapper.setConfig(uiConfig);
```
Comment (Yuvalke) [inline src/kaltura-player.js:284]: keep set config for ui

File: src/common/plugins/plugins-config-store.js [hunk 3]
```diff
@@ -48,6 +48,9 @@ const defaultConfig: dataStoreType = {
   },
   vr: {
     rootElement: '{{domRootElementId}}'
+  },
+  share: {
```
Comment (Yuvalke) [inline src/common/plugins/plugins-config-store.js:52]: remove kaltura page share

File: src/kaltura-player.js [hunk 4]
```diff
@@ -278,9 +272,8 @@ class KalturaPlayer extends FakeEventTarget {
       delete localPlayerConfig.sources;
     }
     this._localPlayer.configure(localPlayerConfig);
-    const uiConfig = config.ui;
+    const uiConfig = configDictionary.ui;
```
Comment (Yuvalke) [inline src/kaltura-player.js:275]: change to config.ui, ui could be empty and still configure the ui wrapper

File: src/common/utils/setup-helpers.js [hunk 5]
```diff
@@ -190,6 +191,26 @@ function isDebugMode(): boolean {
...
 
+/**
+ * get the parameter for start time
+ * @private
+ * @param {KPOptionsObject} options - kaltura player options
+ * @returns {void}
+ */
+function maybeApplyStartTimeQueryParam(options: KPOptionsObject): void {
+  let startTime;
+  if (window.URLSearchParams) {
```
Comment (yairans) [inline src/common/utils/setup-helpers.js:202]: same logic as `isDebugMode`. can be united?

File: src/common/utils/setup-helpers.js [hunk 6]
```diff
@@ -254,10 +244,21 @@ function setLogOptions(options: KPOptionsObject): void {
...
-  name = name.replace(/[[]/, '\\[').replace(/[\]]/, '\\]');
-  const regex = new RegExp('[\\?&]' + name + '=([^&#]*)');
-  const results = regex.exec(location.search);
-  return results === null ? '' : decodeURIComponent(results[1].replace(/\+/g, ' '));
+  const getUrlParamPolyfill = () => {
+    name = name.replace(/[[]/, '\\[').replace(/[\]]/, '\\]');
+    const regex = new RegExp('[\\?&]' + name + '=([^&#]*)');
+    const results = regex.exec(location.search);
+    const isExist = location.search.indexOf(name) > -1;
+    return results === null ? (isExist ? '' : results) : decodeURIComponent(results[1].replace(/\+/g, ' '));
```
Comment (yairans) [inline src/common/utils/setup-helpers.js:252]:
```suggestion
    return results === null ? (isExist ? 'true' : null) : decodeURIComponent(results[1].replace(/\+/g, ' '));
```
this is clearer. 
also can you add tests for this please?

**Review by yairans — CHANGES_REQUESTED** [review]

**Comment by yairans** [comment]
Can you add tests for the query params please? you can override `window.URLSearchParams = null` to test the 2 options

File: src/common/utils/setup-helpers.js (hunk 6 above)
Comment (Yuvalke) [reply src/common/utils/setup-helpers.js:252]: I set empty string because of that is the response from native URLSearchParams

**Review by yairans — APPROVED** [review]

File: flow-typed/types/ui-component.js [hunk 7]
```diff
@@ -1,11 +1,5 @@
 // @flow
-declare type KPUIComponent = {
-  label: string,
-  presets: Array<string>,
-  area: string,
+declare type KPUIComponent = KPPluginUIComponent & {
```
Comment (yairans) [inline flow-typed/types/ui-component.js:2]: all components are plugin component? no, so seems the type is incorrect. KPUIComponentOptions?

**Review by yairans — CHANGES_REQUESTED** [review]

File: flow-typed/types/plugin-ui-component.js [hunk 8]
```diff
@@ -0,0 +1,9 @@
+declare type KPPluginUIComponent = {
```
Comment (yairans) [inline flow-typed/types/plugin-ui-component.js:1]: let's declare all in ui-component.js

**Review by yairans — APPROVED** [review]

**Pull request merged**